src/js/Helpers/Layover.js

```diff
@@ -0,0 +1,868 @@
+    const { offsetWidth, offsetHeight } = child;
+
+    let left;
+    let top;
+    if (x === HorizontalAnchors.CENTER) {
+      const { left: childLeft } = child.getBoundingClientRect();
+      left = (childLeft + (rect.width / 2) - (offsetWidth / 2));
+    } else if (x === HorizontalAnchors.INNER_RIGHT) {
+      left = rect.right - offsetWidth;
+    } else if (x === HorizontalAnchors.LEFT) {
+      left = rect.left - offsetWidth;
+    }
+
+    if (centered && x === HorizontalAnchors.CENTER && y === VerticalAnchors.CENTER) {
+      top = rect.top - (offsetHeight / 2) + (rect.height / 2);
+    } else if (y === VerticalAnchors.TOP) {
+      top = rect.top - offsetHeight;
+    } else if (y === VerticalAnchors.CENTER) {
+      top = rect.top + rect.height / 2;
+    } else if (y === VerticalAnchors.BOTTOM) {
+      top = rect.bottom;
+    }
+
+    const style = {};
+    if (top) {
+      style.top = top;
+    }
+
+    if (left) {
+      style.left = left;
+    }
+
+    return style;
+  }
+
+  /**
+   * Whew. Ok. So since the fixedTo prop can either be two elements or a single item,
+   * this utility function is used to add/remove the scrolling event listeners for
+   * this prop.
+   *
+   * When the fixedTo prop has a horizontal and/or vertical attribute, the `window`
+   * will be the fallback option. If both the horizontal and vertical attributes are
+   * defined, the `window` still needs to have a scroll listener to make sure it
+   * doesn't go off screen.
+   */
+  _manageFixedToListener = (fixedTo, add) => {
+    const listener = `${add ? 'add' : 'remove'}EventListener`;
+    if (fixedTo !== window && (fixedTo.x || fixedTo.y)) {
+      const { x, y } = fixedTo;
+      if (x) {
+        x[listener]('scroll', this._handleScroll);
+      } else {
+        window[listener]('scroll', this._handleScroll);
+      }
+
+      if (y) {
+        y[listener]('scroll', this._handleScroll);
+      } else if (!x) {
+        // Only add the window event listener once
+        window[listener]('scroll', this._handleScroll);
+      }
+
+      if (y && y !== window && x && x !== window) {
+        window[listener]('scroll', this._handleScroll);
+      }
+    } else {
+      fixedTo[listener]('scroll', this._handleScroll);
+
+      if (fixedTo !== window) {
+        window[listener]('scroll', this._handleScroll);
+      }
+    }
+  };
+
+  /**
+   * This is just a simple utility function to merge the existing state styles,
+   * any new styles, and the children's styles (with most precidence).
+   */
+  _mergeStyles = (style) => ({
+    ...this.state.styles,
+    ...style,
+    ...React.Children.only(this.props.children).props.style,
+  });
+
+  /**
+   * This initializes the popover with the default styles, and the intitial bookkeeping
+   * variables to update while it is open.
+   */
+  _init = (fixedTo, anchor, sameWidth, centered, rect) => {
+    const centeredDialog = this._dialog && this._dialog.classList.contains('md-dialog--centered');
```

Comment:
This isn't a real fix since if a developer uses a custom fixed element, the same issue will happen..